Scripts/CI/README_Metadata_StyleCheck/entry.py

```diff
@@ -0,0 +1,564 @@
+import os
+import argparse
+import subprocess as sp
+from os import listdir
+from os.path import isfile, join
+import json
+
+def main():
+    # Tell the program that we're using global variables
+    msg = 'Entry point of the docker to run json style check scripts.'
+    parser = argparse.ArgumentParser(description=msg)
+    parser.add_argument('-s', '--string', help='A JSON array of file paths.')
+    args = parser.parse_args()
+    cleanstring = ""
+    try:
+        for letter in args.string:
+            if letter not in '[" \']':
+                cleanstring+=letter
+        file_list = cleanstring.split(",")
+    except Exception as e:
+        print(f"Unable to split args {args.string}.\nException: {e}")
+        exit(1)
+    check_files(file_list)
+    
+def check_files(file_list):
+    print("Files to check: ")
+    for file in file_list:
+        print(file)
+    print("\n")
+    total_errors = 0
+    if not file_list:
+        print("No files in file set")
+        exit(0)
+    for file in file_list:
+        try:
+            errors = []
+            # Set variables for file
+            directory_list = file.split("/")
+            file_name = directory_list[-1]
+            
+            # Check skip file conditions
+            # Does the file exist, is it in a category, is it a .json or .md?
+            if skip_file(directory_list):
+                continue        
+
+            # If metadata file
+            if file_name.lower() == 'readme.metadata.json':
+                try:
+                    json_file = MetadataFile(file)
+                    errors += json_file.check_metadata_file_for_errors()
+                except Exception as e:
+                    errors.append(f"Error checking {file_name}. Exception: {e}")
+
+            # If README.md
+            elif file_name.lower() == 'readme.md':
+                try:
+                    readme_file = READMEFile(file)
+                    errors += readme_file.check_readme_for_errors()
+                except Exception as e:
+                    errors.append(f"Error checking {file_name}. Exception: {e}")
+            
+            # Count and print the errors
+            if len(errors) > 0:
+                print(f"Found {len(errors)} errors in file:\n{file}")
+                for i in range(len(errors)):
+                    print(f"{i+1}. {errors[i]}")
+                print("\n")
+            total_errors += len(errors)
+        except Exception as e:
+            print(f"Critical failure on file: {file}. Exception: {e}")
+            total_errors += 1
+    
+    if total_errors == 0:
+        print("All tests passed!")
+    else:
+        print(f"Total errors: {total_errors}")
+    # When we exit an pass a non-zero value, GitHub registers the test as failed.
+    exit(total_errors)
+
+def skip_file(directory_list: list)-> bool:
```

Comment:
Not required, but I'd consider a different name. Skip file seems like a method that performs some action, but really it is checking if valid and determining if you should skip the file. I find it more useful to actually flip around the function to be `isFileValid` and return True if it meets all conditions or False if not.